include/swift/AST/DiagnosticsSema.def

```diff
@@ -6312,9 +6315,8 @@ NOTE(objc_witness_objc_requirement,none,
 ERROR(no_opaque_return_type_of,none,
       "unable to resolve type for _opaqueReturnTypeOf attribute", ())
 
-
 ERROR(objc_observing_accessor, none,
-      "observing accessors are not allowed to be marked @objc", ())
+      "%0 is not allowed to be marked '@objc'", (DescriptiveDeclKind))
```

Comment:
This is another instance of trading generality for but a slightly more precise description of what the diagnostic is already pointing at, which I believe is not a fair exchange. I can think of 3 ways to both retain generality and get rid of the "accessor" term.

1. Replace the template parameter with "observers", [as they are called in the book](https://docs.swift.org/swift-book/documentation/the-swift-programming-language/declarations).
1. Attach a note articulating the general rule. This is mostly text duplication because the rule hasn’t gotten any simpler— far from ideal.
1. `observers (%0) are not allowed ...`.

Reply:
Thoughts?

Reply:
agreed, i applied third suggestion here: 
618051a3edf

i think it makes more sense to let the diagnostics message give more information in the case, because ad a swift-user i would get that it's about all observers